### PR TITLE
Conviva API Update : To Be Deleted

### PR DESCRIPTION
**Pull request by realeyes-mike-patterson**
This PR Covers
* Using the new 4.0 Conviva API
* Reporting Average FPS when a session ends.

### CODE REVIEW

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java [hunk 1]
```diff
@@ -43,42 +44,35 @@ public void setPlaybackStarted(boolean value) {
...
             }
 
-            contentMetadata.viewerId = getViewerId();
-            contentMetadata.streamType = ObjectUtils.defaultIfNull(
-                    metadataOverrides.getStreamType(),
-                    metadata.getStreamType());
-
-            contentMetadata.applicationName = ObjectUtils.defaultIfNull(
-                    metadataOverrides.getApplicationName(),
-                    metadata.getApplicationName());
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java:60]: ~What happened to `StreamType` and `ApplicationName`? Are they no longer needed? 🤔~

EDIT: found them 🙂 👍

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java [hunk 2]
```diff
@@ -99,10 +93,22 @@ public String getViewerId() {
         return ObjectUtils.defaultIfNull(metadataOverrides.getViewerId(), metadata.getViewerId());
     }
 
-    public void setStreamType(ContentMetadata.StreamType newValue) {
+    public ConvivaSdkConstants.StreamType getStreamType() {
+        return ObjectUtils.defaultIfNull(
+                metadataOverrides.getStreamType(),
+                metadata.getStreamType());
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java:99]: Nit: code style
```suggestion
        return ObjectUtils.defaultIfNull(
            metadataOverrides.getStreamType(),
            metadata.getStreamType()
        );
```

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java [hunk 3]
```diff
@@ -99,10 +93,22 @@ public String getViewerId() {
...
+    }
+
+    public void setStreamType(ConvivaSdkConstants.StreamType newValue) {
         metadata.setStreamType(newValue);
     }
 
+    public String getApplicationName() {
+        return ObjectUtils.defaultIfNull(
+                metadataOverrides.getApplicationName(),
+                metadata.getApplicationName());
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java:109]: Nit: code style
```suggestion
        return ObjectUtils.defaultIfNull(
            metadataOverrides.getApplicationName(),
            metadata.getApplicationName()
        );
```

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java [hunk 4]
```diff
@@ -89,28 +84,20 @@ public ConvivaAnalytics(BitmovinPlayer player,
...
+        Map<String, Object> convivaSettings = new HashMap<>();
+        if (config.isDebugLoggingEnabled()) {
+            convivaSettings.put(ConvivaSdkConstants.LOG_LEVEL, ConvivaSdkConstants.LogLevel.DEBUG);
+        }
+        if (config.getGatewayUrl() != null) {
+            convivaSettings.put(ConvivaSdkConstants.GATEWAY_URL, config.getGatewayUrl());
+        }
 
-            this.client = new Client(clientSettings, androidSystemFactory);
+        com.conviva.sdk.ConvivaAnalytics.init(context, customerKey, convivaSettings);
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:98]: Did they really change the type name to the same we have 😂

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java [hunk 5]
```diff
@@ -217,30 +196,28 @@ public void updateContentMetadata(MetadataOverrides metadataOverrides) {
      * Sends a custom deficiency event during playback to Conviva's Player Insight. If no session is active it will NOT
      * create one.
      *
-     * @param message Message which will be send to conviva
+     * @param message  Message which will be send to conviva
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:199]: nit: We usually don't align the properties in the documentation but fine 4 me 🙂

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java [hunk 6]
```diff
@@ -291,54 +259,60 @@ private void customEvent(BitmovinPlayerEvent event, Map<String, Object> attribut
...
 
+    private ConvivaExperienceAnalytics.ICallback videoAnalyticsCallback = new ConvivaExperienceAnalytics.ICallback() {
+        @Override
+        public void update() {
+        }
+
+        @Override
+        public void update(String s) {
+        }
+    };
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:287]: Do we need this if we don't implement anything in there? 🤔

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java [hunk 7]
```diff
@@ -291,54 +259,60 @@ private void customEvent(BitmovinPlayerEvent event, Map<String, Object> attribut
...
+            convivaVideoAnalytics.reportPlaybackMetric(ConvivaSdkConstants.PLAYBACK.RESOLUTION, videoQuality.getWidth(), videoQuality.getHeight());
+            convivaVideoAnalytics.reportPlaybackMetric(ConvivaSdkConstants.PLAYBACK.RENDERED_FRAMERATE, Math.round(videoQuality.getFrameRate()));
         }
 
-        try {
-            client.updateContentMetadata(sessionId, contentMetadataBuilder.build());
-        } catch (ConvivaException e) {
-            Log.e(TAG, e.getLocalizedMessage());
-        }
+//        convivaVideoAnalytics.reportPlaybackMetric(contentMetadataBuilder.getStreamType().toString());
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:314]: Please remove commented-out code or implement properly (if still needed)

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java [hunk 8]
```diff
@@ -631,44 +591,35 @@ public void onTimeShifted(TimeShiftedEvent timeShiftedEvent) {
...
-        try {
-            playerStateManager.setPlayerSeekStart(seekTarget);
-        } catch (ConvivaException e) {
-            Log.e(TAG, e.getLocalizedMessage());
-        }
-    };
+        convivaVideoAnalytics.reportPlaybackMetric(ConvivaSdkConstants.PLAYBACK.SEEK_STARTED, seekTarget);
+    }
+
+    ;
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:603]: Please remove this lines 🙂

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java [hunk 9]
```diff
@@ -65,13 +60,13 @@
...
-    private Client client;
     private BitmovinPlayer bitmovinPlayer;
     private ContentMetadataBuilder contentMetadataBuilder = new ContentMetadataBuilder();
     private ConvivaConfiguration config;
-    private int sessionId = Client.NO_SESSION_KEY;
-    private PlayerStateManager playerStateManager;
+    private Context context;
+    private ConvivaVideoAnalytics convivaVideoAnalytics;
     private MetadataOverrides metadataOverrides;
+    private boolean activeSession = false;
```
Comment (stonko1994) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:70]: Just to double-check: the Ad tracking works differently than on iOS? No need for an `adAnalytics` object? 🤔

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java (hunk 4 above)
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:98]: Yes. I can make the change to our name but then there will be ripple effects like documentation, release notes, breaking changes for clients, and a difference from iOS.

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java (hunk 9 above)
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:69]: Correct. There's an option to add more ad info but this matches what the previous integration was doing. Here's a snapshot of touchstone with this PR implemented:
Snip20210614_152

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java (hunk 6 above)
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java]: Removed

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java (hunk 8 above)
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java]: Done - whoops

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java (hunk 7 above)
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java]: Done - whoops

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java (hunk 3 above)
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java]: Done

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java (hunk 2 above)
Comment (realeyes-mike-patterson) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ContentMetadataBuilder.java]: Done

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java (hunk 4 above)
Comment (stonko1994) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:97]: No this is totally fine 🙂

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java (hunk 9 above)
Comment (stonko1994) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java:69]: 👍

**Review by stonko1994 — APPROVED** [review]
LGTM 👍

**Comment by realeyes-mike-patterson** [comment]
> LGTM 👍

Do you want the minor update #30 merged first?

**Comment by luckygoyal-bitmovin** [comment]
> > LGTM 👍
> 
> Do you want the minor update #30 merged first?

@realeyes-mike-patterson , yes would be better to merge #30 before this.

**Comment by luckygoyal-bitmovin** [comment]
Superseded by another PR and commit for Conviva API upgrade done by Eyevinn team. Closing this PR as obsolete.

**Pull request closed without merging**